### PR TITLE
Add ability to detect if running on `VMware` system

### PR DESCRIPTION
**Pull request by jamie-suse**

### CODE REVIEW

**Comment by jamie-suse** [comment]
While making this PR, I found I was writing a lot of repeated code. Thoughts on a possible refactor?

File: internal/core/cloud/metadata.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ const (
 	GCP     = "gcp"
 	Nutanix = "nutanix"
 	KVM     = "kvm"
+	VMWARE  = "vmware"
```
Comment (arbulu89) [inline internal/core/cloud/metadata.go:17]: Maybe bettter `VMware` as the constant name

**Review by arbulu89 — COMMENTED** [review]
If we want to reduce the amount of times the `systemd-detect-virt` command is run, we could maybe create a new function like `systemdDetectVirt` which returns the command output value and then:

1. have some map like `{kvm: true, vmware: true}` or similar. Once the command is executed you could look up in the map, and if there is a value you know which is the detected one. In the other hand, you could return nil.

2. or simply continue having `identifyKVM` and `identifyVMware` functions, but using the command output as parameter and simply do the comparison there.

Take some time to think about other solutions as well.
Anyway, the current code should work, and even with high level of duplicated code, I find it easy to navigate

**Review by CDimonaco — COMMENTED** [review]
Agree with @arbulu89 the code duplication and the command execution is really duplicated, the code could be a lot improved even in term of readability.

BUT, this is not the scope of this pr, the code works so we can think about later :D

**Review by rtorrero — APPROVED** [review]
Code looks good, green light from my side and we can later tackle how to reduce the duplication of code.

**Review by CDimonaco — APPROVED** [review]

**Pull request merged**